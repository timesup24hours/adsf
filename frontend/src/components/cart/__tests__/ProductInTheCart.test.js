// npm packages
import configureMockStore from 'redux-mock-store'
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import Dialog from 'material-ui/Dialog';

// our packages
import ProductInTheCartWrapper, { ProductInTheCart } from '../ProductInTheCart'

// create mock store
const mockStore = configureMockStore();

const cart = {
  cart: [],
  totalQuantity: 0,
  total: 0,
  pending: false,
  success: false,
  fail: false,
  errors: {
  },
  error: false,
  currentProduct: {},
  quantityOfCurrentProduct: 0,
}


const product = {
  images: [],
  _id: '1',
  name: 'asd',
  price: 123,
  soldBy: 'asdf',
}

const product1 = {
  product: product,
  total: 1,
  quantity: 1,
}

const product2 = {
  product: product,
  total: 1,
  quantity: 2,
}

const subtractQuantityOfProductIntheCart = payload => expect(payload.productId).toBe(product1.product._id)
const increaseQuantityOfProductInTheCart = payload => expect(payload.productId).toBe(product1.product._id)
const removeProductFromCart = payload => expect(payload.productId).toBe(product1.product._id)
const totalQuantity = 1


const store = mockStore({ cart: totalQuantity });

test('# ProductInTheCartWrapper', () => {
  const wrapper = shallow(<ProductInTheCartWrapper store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# ProductInTheCart', () => {
  const wrapper = shallow(<ProductInTheCart
    subtractQuantityOfProductIntheCart={subtractQuantityOfProductIntheCart}
    increaseQuantityOfProductInTheCart={increaseQuantityOfProductInTheCart}
    removeProductFromCart={removeProductFromCart}
    totalQuantity={totalQuantity}
    product={product1}
     />);

  expect(wrapper).toMatchSnapshot();
});

test('# ProductInTheCart should render nested div', () => {
  const component = (
    <ProductInTheCart
      subtractQuantityOfProductIntheCart={subtractQuantityOfProductIntheCart}
      increaseQuantityOfProductInTheCart={increaseQuantityOfProductInTheCart}
      removeProductFromCart={removeProductFromCart}
      totalQuantity={totalQuantity}
      product={product2}
      store={store}
       />
  )

  const app = shallow(component);
  expect(app).toMatchSnapshot();

});

test('# ProductInTheCart should render nested div', () => {
  const component = (
    <ProductInTheCart
      subtractQuantityOfProductIntheCart={subtractQuantityOfProductIntheCart}
      increaseQuantityOfProductInTheCart={increaseQuantityOfProductInTheCart}
      removeProductFromCart={removeProductFromCart}
      totalQuantity={totalQuantity}
      product={product2}
   />
  )
  const app = mount(component, {
      context: {
        muiTheme: getMuiTheme(),
      },
      childContextTypes: {
        muiTheme: React.PropTypes.object.isRequired,
      },
    });

  app.find('#ProductInTheCart-substract').hasClass('ProductInTheCart-info-quantity-change')
});

test('# ProductInTheCart simulate click', () => {
  const component = (
    <ProductInTheCart
      subtractQuantityOfProductIntheCart={subtractQuantityOfProductIntheCart}
      increaseQuantityOfProductInTheCart={increaseQuantityOfProductInTheCart}
      removeProductFromCart={removeProductFromCart}
      totalQuantity={totalQuantity}
      product={product2}
   />
  )
  const app = mount(component, {
      context: {
        muiTheme: getMuiTheme(),
      },
      childContextTypes: {
        muiTheme: React.PropTypes.object.isRequired,
      },
    });

  // const mountWithTheme = (node) => mount(<MuiThemeProvider>{node}</MuiThemeProvider>);
  // const app = mount(mountWithTheme(component))

  app.find('#ProductInTheCart-substract').simulate('click');
  app.find('#ProductInTheCart-add').simulate('click')
  app.find('#ProductInTheCart-open-dialog').simulate('click')
  expect(app.state().open).toBe(true)

});
