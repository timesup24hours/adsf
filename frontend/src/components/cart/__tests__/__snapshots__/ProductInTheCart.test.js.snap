exports[`test # ProductInTheCart 1`] = `
<div
  className="ProductInTheCart-container">
  <div
    className="ProductInTheCart-info-container">
    <div
      className="ProductInTheCart-img">
      <img
        alt=""
        src="/images/products/undefined" />
    </div>
    <div
      className="ProductInTheCart-info-holder">
      <div
        className="ProductInTheCart-div ProductInTheCart-info-name">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/product/1">
          asd
        </Link>
      </div>
      <div
        className="ProductInTheCart-div ProductInTheCart-info-price">
        $
        123
      </div>
      <div
        className="ProductInTheCart-div ProductInTheCart-info-quantity-section">
        <div
          className="ProductInTheCart-info-quantity">
          <strong>
            Quantity:
          </strong>
           
          1
        </div>
        <div
          className="ProductInTheCart-info-quantity-change"
          id="ProductInTheCart-add"
          onClick={[Function]}>
          Add
        </div>
      </div>
      <div
        className="ProductInTheCart-info-seller text-flow">
        Sold by : 
        asdf
      </div>
    </div>
  </div>
  <div
    className="ProductInTheCart-right-session">
    <button
      className="waves-effect waves-light btn pink"
      id="ProductInTheCart-open-dialog"
      onClick={[Function]}>
      Remove
    </button>
  </div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          label="No"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
        <FlatButton
          disabled={false}
          id="ProductInTheCart-remove"
          keyboardFocused={true}
          label="Yes"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    id="ProductInTheCart-Dialog"
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Remove and Redirect Confirmation">
    <div>
      Are you want to remove the last item from the cart, and Redirect to the Shop Page
    </div>
  </Dialog>
</div>
`;

exports[`test # ProductInTheCart should render nested div 1`] = `
<div
  className="ProductInTheCart-container">
  <div
    className="ProductInTheCart-info-container">
    <div
      className="ProductInTheCart-img">
      <img
        alt=""
        src="/images/products/undefined" />
    </div>
    <div
      className="ProductInTheCart-info-holder">
      <div
        className="ProductInTheCart-div ProductInTheCart-info-name">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/product/1">
          asd
        </Link>
      </div>
      <div
        className="ProductInTheCart-div ProductInTheCart-info-price">
        $
        123
      </div>
      <div
        className="ProductInTheCart-div ProductInTheCart-info-quantity-section">
        <div
          className="ProductInTheCart-info-quantity">
          <strong>
            Quantity:
          </strong>
           
          2
        </div>
        <div
          className="ProductInTheCart-info-quantity-change"
          id="ProductInTheCart-add"
          onClick={[Function]}>
          Add
        </div>
        <div
          className="ProductInTheCart-info-quantity-change"
          id="ProductInTheCart-substract"
          onClick={[Function]}>
          Substract
        </div>
      </div>
      <div
        className="ProductInTheCart-info-seller text-flow">
        Sold by : 
        asdf
      </div>
    </div>
  </div>
  <div
    className="ProductInTheCart-right-session">
    <button
      className="waves-effect waves-light btn pink"
      id="ProductInTheCart-open-dialog"
      onClick={[Function]}>
      Remove
    </button>
  </div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          label="No"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
        <FlatButton
          disabled={false}
          id="ProductInTheCart-remove"
          keyboardFocused={true}
          label="Yes"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    id="ProductInTheCart-Dialog"
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Remove and Redirect Confirmation">
    <div>
      Are you want to remove the last item from the cart, and Redirect to the Shop Page
    </div>
  </Dialog>
</div>
`;

exports[`test # ProductInTheCartWrapper 1`] = `
<ProductInTheCart
  increaseQuantityOfProductInTheCart={[Function]}
  removeProductFromCart={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  subtractQuantityOfProductIntheCart={[Function]} />
`;
